examples/md-flexible/src/Simulation.cpp

```diff
@@ -120,11 +120,23 @@ Simulation::Simulation(const MDFlexConfig &configuration,
   _autoPasContainer->setAllowedCellSizeFactors(*_configuration.cellSizeFactors.value);
   _autoPasContainer->setAllowedContainers(_configuration.containerOptions.value);
 
-  if (_configuration.getInteractionTypes().size() == 0) {
-    _configuration.functorOption.value = MDFlexConfig::FunctorOption::lj12_6;
+  if (_configuration.getInteractionTypes().empty()) {
+    std::string functorName{};
+    _configuration.functorOption.value =
+#if defined(MD_FLEXIBLE_FUNCTOR_AVX) && defined(__AVX__)
+        MDFlexConfig::FunctorOption::lj12_6_AVX;
+    functorName = "Lennard-Jones AVX Functor.";
+#elif defined(MD_FLEXIBLE_FUNCTOR_SVE) && defined(__ARM_FEATURE_SVE)
+        MDFlexConfig::FunctorOption::lj12_6_SVE;
+    functorName = "Lennard-Jones SVE Functor.";
+#else
+        MDFlexConfig::FunctorOption::lj12_6;
+    functorName = "Lennard-Jones AutoVec Functor.";
+#endif
```

Comment:
Not entirely sure if this is better so decide for yourself. I didn't like the macro being a mixture of a splitted declaration + regular assignment.

```c++
    std::string functorName{};
    std::tie(_configuration.functorOption.value, functorName) =
#if defined(MD_FLEXIBLE_FUNCTOR_AVX) && defined(__AVX__)
        std::make_tuple(MDFlexConfig::FunctorOption::lj12_6_AVX, "Lennard-Jones AVX Functor.");
#elif defined(MD_FLEXIBLE_FUNCTOR_SVE) && defined(__ARM_FEATURE_SVE)
        std::make_tuple(MDFlexConfig::FunctorOption::lj12_6_SVE, "Lennard-Jones SVE Functor.");
#else
        std::make_tuple(MDFlexConfig::FunctorOption::lj12_6, "Lennard-Jones AutoVec Functor.");
#endif
```

Reply:
No strong opinion here, so I changed it to your suggestion [484cae6](https://github.com/AutoPas/AutoPas/pull/751/commits/484cae676be2ab1996c0132aa654fc56cff078bf)